src/Components/Modals/EditModal/EditModal.tsx

```diff
@@ -0,0 +1,89 @@
+import './EditModal.scss';
+import {Movie} from '../../App';
+import {useState} from 'react';
+import DatePicker from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
+
+interface EditModalProps{
+  movie?: Movie,
+  onClose: () => void
+}
+
+const genres = ['Action & Adventure', 'Drama, Biographt, Music', 'Oscar winning Movie'];
+
+export function EditModal({onClose, movie} : EditModalProps) {
+  const isNew = movie === null;
+
+  const [title, setTitle] = useState(movie && movie.title || '');

```

Comment:
Because of you work with complex state instead of using complex state I would use useReducer here
Look at https://ru.reactjs.org/docs/hooks-reference.html#usereducer
Also as variant you could use https://react-hook-form.com for working with forms